Army/units/unit.h

```diff
@@ -0,0 +1,79 @@
+#ifndef UNIT_H
+#define UNIT_H
+
+#include <iostream>
+#include "../combat/ability.h"
+
+using namespace std;
+
+class Ability;
+class State;
+
+class UnitIsDeadException {};
+class IsSelfAttackException {};
+class ManaIsOverException {};
+class IsVictimException {};
+class IsFriendlyAttackException {};
+
+enum UnitType {
+    soldierType,
+    rogueType,
+    berserkType,
+    vampireType,
+    werewolfType,
+    wizardType
+};
+
+class Unit {
+    private:
+        string name;
+        int healthPointLimit;
+        int currentHP;
+        int damage;
+        
+    protected:
+        bool isFriendly;
+        int unitType;
+        State* normalState;
+        State* wolfState;
+        Ability* ability;
+
+        virtual void ensureIsAlive();
+        virtual void ensureIsNotSelfAttack(Unit* victim);
+        virtual void ensureIsNotAlly(Unit* target);
+
+    public:
+        Unit(const string& name, int healthPoint, int damage);
+        virtual ~Unit();
```

Comment:
You keep pointers to objects inside (`State* normalState`, `State* wolfState`, `Ability* ability`), but don't remove them. Looks like memory leak